src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -96,24 +34,94 @@ const defaultProps = {
     isVisible: false,
 };
 
-const ReportActionContextMenu = (props) => {
-    const wrapperStyle = getReportActionContextMenuStyles(props.isMini);
-    return props.isVisible && (
-        <View style={wrapperStyle}>
-            {CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
-                <ReportActionContextMenuItem
-                    icon={contextAction.icon}
-                    text={contextAction.text}
-                    successIcon={contextAction.successIcon}
-                    successText={contextAction.successText}
-                    isMini={props.isMini}
-                    onPress={() => contextAction.onPress(props.reportAction)}
-                    key={contextAction.text}
-                />
-            ))}
-        </View>
-    );
-};
+class ReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        /**
+         * A list of all the context actions in this menu.
+         */
+        this.CONTEXT_ACTIONS = [
+            // Copy to clipboard
+            {
+                text: 'Copy to Clipboard',
+                icon: ClipboardIcon,
+                successText: 'Copied!',
+                successIcon: Checkmark,
+                shouldShow: true,
+
+                // If return value is true, we switch the `text` and `icon` on
+                // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+                // the `text` and `icon`
+                onPress: () => {
+                    const message = _.last(lodashGet(this.props.reportAction, 'message', null));
+                    const html = lodashGet(message, 'html', '');
+                    const text = lodashGet(message, 'text', '');
+                    const isAttachment = _.has(this.props.reportAction, 'isAttachment')
+                        ? this.props.reportAction.isAttachment
+                        : isReportMessageAttachment(text);
+                    if (!isAttachment) {
+                        Clipboard.setString(text);
+                    } else {
+                        Clipboard.setString(html);
+                    }
+                },
+            },
+
+            // Copy chat link
+            {
+                text: 'Copy Link',
+                icon: LinkCopy,
+                shouldShow: false,
+                onPress: () => {},
+            },
+
+            // Mark as Unread
+            {
+                text: 'Mark as Unread',
+                icon: Mail,
+                shouldShow: false,
+                onPress: () => {},
+            },
+
+            // Edit Comment
+            {
+                text: 'Edit Comment',
+                icon: Pencil,
+                shouldShow: false,
+                onPress: () => {},
+            },
+
+            // Delete Comment
+            {
+                text: 'Delete Comment',
+                icon: Trashcan,
+                shouldShow: true,
```

Comment:
I think we need to be careful to make sure that this only shows for report actions authored by the current user. To do that, I think you can add a `session` prop and wrap this component in the `withOnyx` HOC to bind the `ONYXKEYS.SESSION` key to the `session` prop. Then update this `shouldShow` like:

```suggestion
                shouldShow: this.props.reportAction.actorEmail === this.props.session.email,
```

Reply:
+1